packages/core/src/utils/Util.ts

```diff
@@ -1021,3 +1023,31 @@ export const getValueAsShapeList = <T>(val: Value<T>): Shape<T>[] => {
 };
 
 //#endregion
+
+//#region functions
+
+export const noWarn = <T>(value: T): MayWarn<T> => ({
+  value,
+  warnings: [],
+});
+
+export const noWarnFn = <T extends any[], S>(
+  f: (...args: T) => S
+): ((...args: T) => MayWarn<S>) => {
+  return (...args: T) => noWarn(f(...args));
+};
+
+//#endregion
+
+//#region warnings
+
+export const allWarnings: Warning["tag"][] = [
```

Comment:
Possible hack to avoid this list: https://stackoverflow.com/questions/44480644/string-union-to-string-array?

Reply:
Did this instead:
```ts
export const allWarnings = [
  "BBoxApproximationWarning",
  "GroupCycleWarning",
  "ImplicitOverrideWarning",
  "LayerCycleWarning",
  "NoopDeleteWarning",
  "ShapeBelongsToMultipleGroups",
] as const;

// These are type-level assertions that allWarnings
// covers each variant in StyleWarning
type AssertedWarningTags = (typeof allWarnings)[number];
type ActualWarningTags = StyleWarning["tag"];

type IsSubset<T, U> = T extends U ? true : false;
type AreUnionsEqual<T, U> = IsSubset<T, U> extends true
  ? IsSubset<U, T>
  : false;

// If this fails, then allWarnings and the actual tags of StyleWarning variants
// are different.
const _warningTagsCheck: AreUnionsEqual<
  AssertedWarningTags,
  ActualWarningTags
> = true;

```